sockeye/translate.py

```diff
@@ -56,6 +56,13 @@ def run_translate(args: argparse.Namespace):
         check_condition(args.beam_size == 1, "--skip-topk has no effect if beam size is larger than 1")
         check_condition(len(args.models) == 1, "--skip-topk has no effect for decoding with more than 1 model")
 
+    if args.nbest_size > 1:
+        check_condition(args.beam_size >= args.nbest_size,
+                        "Size of nbest list (--nbest-size) must be smaller or equal to beam size (--beam-size).")
+        if args.output_type not in C.OUTPUT_HANDLERS_NBEST:
+            logger.warning("For nbest translation, output handler must be one of %s, overriding option --output-type to '%s'.",
+                       C.OUTPUT_HANDLERS_NBEST, C.OUTPUT_HANDLER_NBEST)
+
```

Comment:
See comment above about just applying the existing output handler to every item in an n-best list.